src/ucp/rndv/rndv_rkey_ptr.c

```diff
@@ -218,11 +218,14 @@ ucp_proto_rndv_rkey_ptr_fetch_progress(uct_pending_req_t *uct_req)
 
 static ucs_status_t ucp_proto_rndv_rkey_ptr_reset(ucp_request_t *request)
 {
+    ucp_datatype_iter_t *dt_iter = &request->send.state.dt_iter;
+
     switch (request->send.proto_stage) {
     case UCP_PROTO_RNDV_RKEY_PTR_STAGE_COPY:
         ucs_queue_remove(&request->send.ep->worker->rkey_ptr_reqs,
                          &request->send.rndv.rkey_ptr.queue_elem);
         ucp_proto_rndv_rkey_destroy(request);
+        dt_iter->offset = dt_iter->length;
```

Comment:
IMO we need to do this when resetting in the ACK stage. Otherwise how would the next protocol finish the copy operation?

Reply:
The next proto is RTR, which will trigger rndv_put with offset = 0

Reply:
what is the problem with doing rndv_put with offset=0?